tests/db_migrator_input/config_db/reclaiming-buffer-warmreboot-expected.json

```
@@ -2043,6 +2043,6 @@
         "admin_status": "up"
     },
     "VERSIONS|DATABASE": {
-        "VERSION": "version_3_0_5"
+        "VERSION": "version_3_0_6"
```

```suggestion
        "VERSION": "version_3_0_3"
```
We should not update db version in `expected` config_db every time a new db version is introduced.
There is an infrastructure to tolerate the case that the target database version in `expected` config_db file isn't the latest database version. The logic is to advance the DB version in the `expected` config_db iff it is the latest version that the feature handles. Check `advance_version_for_expected_database` for details.
Reclaiming buffer unit test does leverage this infra, so theoretically, we should not touch it every time.
However, the version was somehow updated by some previous PR, which breaks the logic and requires the version to be updated every time a new db version is introduced, which is inconvenient to developers.
Now that you are going to introduce new db version, let's change it back to the original `expected` version.

---

Done, thanks